### PR TITLE
Misc_improvements

### PR DESCRIPTION
**Pull request by DaneelTrevize**
Correct the unifying label width mechanism, for "Map Tier" being longer than "R. Level".

Zana's new signature mod is very verbosely worded. The QCompleter mods dropdown forces the min-width of the search UI to the longest mod string in the list. So I replaced a few Master mods with* prefixed shorter names. More experience with Qt would be required to possibly find a way to alter the popup's width behaviour via QAbstractItemView, and/or styling these placeholder mod lines (e.g. with italics). An example of the new minimum size of the mods dropdown can be seen [here](https://i.imgur.com/nu9pwGi.png).  
Fix use of disable_login bool for certain Login DisplayErrors.  
Add support for "% Increased Attack and Cast speed" mod (found on jewels), increased spell base crit chance, and % maximum life.  
Subjectively improved the order of mods in the dropdown filter.  
Tidy some declaration orders, NULL pointers, unused parameters to reduce warnings.

### CODE REVIEW

File: src/buyoutmanager.cpp [hunk 1]
```diff
@@ -614,7 +614,7 @@ std::vector<CurrencyType> Currency::Types() {
 }
 
 Currency Currency::FromIndex(int index) {
-    if (index >= currency_type_as_tag_.size()) {
+    if (static_cast<unsigned int>(index) >= currency_type_as_tag_.size()) {
```
Comment (xyzz) [inline src/buyoutmanager.cpp]: Use size_t instead if you want to cast, also it needs an additional negative check.

**Comment by DaneelTrevize** [comment]
Again, haven't touched C++ in ~15 years, mostly just went with what the suggested Qt Creator and googling recent C++11 discussions pointed me to, in order to reduce the warnings.

I guess I now modify the branch, and then bump this PR with a comment when I think it's ready for another review?

File: src/buyoutmanager.cpp [hunk 2]
```diff
@@ -496,7 +496,7 @@ bool Buyout::IsValid() const {
     case BUYOUT_TYPE_INHERIT:
     case BUYOUT_TYPE_NO_PRICE:
         return true;
-        break;
+        //break;
```
[the remark this answers is not in the document]
Comment (DaneelTrevize) [reply src/buyoutmanager.cpp]: For the same reason that the unreachable statement was there in the first place? To remind future editors to put a break in if they change the logic? Removing it's fine by me.

**Comment by xyzz** [comment]
Yeah that sounds fine.

**Comment by ericsium** [comment]
You can make another commit and then squash the new commit in with a previous and re-edit the comment if you want.  Once you update the branch on github servers it will re-qualify the change and we'll see the update in this thread.

You can do this using git 'rebase' command.  It allows you to manipulate the last X checkins on a branch, re-order them, update comments, squash them together etc... It's a good thing to get familar with.

For example, you have 2 commits on this branch, if you checked in a third commit you could run 'git rebase -i HEAD~3'.

If you're rebased a local branch you need to push to server with 'git push -f' to force the push.  Once that's done your pull will be automatically re-qualified.

File: src/bucket.cpp [hunk 3]
```diff
@@ -14,14 +14,19 @@ void Bucket::AddItem(const std::shared_ptr<Item> & item) {
...
 const std::shared_ptr<Item> &Bucket::item(int row) const
-{   
-    if (row < 0 || row >= items_.size()) {
-        QMessageBox::critical(0, "Fatal Error", QString("Item row out of bounds: ") +
-                              QString::number(row) + " item count: " + QString::number(items_.size()) +
-                              ". Program will abort.");
-        abort();
+{
+    if (row >= 0) {
+        std::vector<Items>::size_type row_t = static_cast<std::vector<Items>::size_type>(row);  // Assumes int max() always able to fit in unsigned long long
```
[the remark this answers is not in the document]
Comment (DaneelTrevize) [reply src/bucket.cpp]: > warning: use of old-style cast

[the remark this answers is not in the document]
Comment (DaneelTrevize) [reply src/bucket.cpp]: ::size_type is also already used in /src/item.ccp:71

**Comment by DaneelTrevize** [comment]
Ok, I think I'm done squashing and deduplicating commit comments, review PR again please.

Personally I am against editing history and losing change granularity, especially for remotely pushed changes, but done as requested, I think.

**Comment by DaneelTrevize** [comment]
Going to recreate this PR with a manually rebased branch, because of 498 being closed/consistent policy of rewriting commit history.

**Pull request closed without merging**